### DISCUSSION TITLE
Cortex Engines should be semantically named?

### DISCUSSION THREAD

**Discussion by dan-menlo** [janhq/nitro#1167] [Team Discussions]
## Problem

- Right now, our engines are named `cortex.llamacpp`, `cortex.tensorrt-llm`, and `cortex.onnx`
- We also have `cortex.python`, which is not an engine but more of a runtime
- This is potentially confusing to Devs, as these are the Github repos
- This is also confusing to Users

## Alternatives

- Semantic naming: e.g. `llamacpp-engine`, `tensorrtllm-engine`, `onnx-engine`
- For ONNX, should it be ONNXRuntime instead of ONNX? 
- Is "*-runtime" better? (or should we stick to our nomenclature of 
- What should we call `python-engine`? (it's not really an engine, but more of a Extension base template?)

## Considerations

- Will we encounter any issues if we make the changes now? (e.g. any legacy users who depend on old naming/URL structure?)
- Should we consider all of them as extensions
    - There is a base C++ Extension format
    - There is a base Python Extension format (that packages its own python binary, i.e. `cortex.python`)

**Comment by vansangpfiev** [comment]
- If we only rename the engines, not the repositories, it will not break anything cc: @hiento09 
- @dan-homebrew Could you please elaborate more on `Should we consider all of them as extensions`?
